File: frontend/lib/models/notificacao.dart
enum TipoNotificacao {
  BAIXO_DESEMPENHO,
  BAIXA_FREQUENCIA,
  DESEMPENHO_E_FREQUENCIA,
}

enum StatusNotificacao { PENDENTE, LIDA, ENCAMINHADA }

class Notificacao {
  final int id;
  final int alunoId;
  final String nomeAluno;
  final String matriculaAluno;
  final String nomeTurma;
  final String conteudoIA;
  final String resumo;
  final TipoNotificacao tipo;
  final StatusNotificacao status;
  final DateTime criadaEm;
  final DateTime? encaminhadaEm;

  Notificacao({
    required this.id,
    required this.alunoId,
    required this.nomeAluno,
    required this.matriculaAluno,
    required this.nomeTurma,
    required this.conteudoIA,
    required this.resumo,
    required this.tipo,
    required this.status,
    required this.criadaEm,
    this.encaminhadaEm,
  });

  factory Notificacao.fromJson(Map<String, dynamic> j) => Notificacao(
    id: j['id'],
    alunoId: j['alunoId'] ?? 0,
    nomeAluno: j['nomeAluno'] ?? '',
    matriculaAluno: j['matriculaAluno'] ?? '',
    nomeTurma: j['nomeTurma'] ?? '',
    conteudoIA: j['conteudoIA'] ?? '',
    resumo: j['resumo'] ?? '',
    tipo: TipoNotificacao.values.firstWhere(
      (e) => e.name == j['tipo'],
      orElse: () => TipoNotificacao.BAIXO_DESEMPENHO,
    ),
    status: StatusNotificacao.values.firstWhere(
      (e) => e.name == j['status'],
      orElse: () => StatusNotificacao.PENDENTE,
    ),
    criadaEm: DateTime.parse(j['criadaEm']),
    encaminhadaEm: j['encaminhadaEm'] != null
        ? DateTime.parse(j['encaminhadaEm'])
        : null,
  );
}

File: backend/src/main/java/com/fateczl/sistemaDeGestaoEscolar/notificacao/Notificacao.java
package com.fateczl.sistemaDeGestaoEscolar.notificacao;

import java.time.LocalDateTime;

import com.fateczl.sistemaDeGestaoEscolar.usuario.aluno.Aluno;
import com.fateczl.sistemaDeGestaoEscolar.usuario.funcionario.Funcionario;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "notificacao")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(of = "id")
public class Notificacao {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "aluno_id", nullable = false)
    private Aluno aluno;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "coordenador_id", nullable = false)
    private Funcionario coordenador;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String conteudoIA;

    @Column(length = 300)
    private String resumo;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private TipoNotificacao tipo;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private StatusNotificacao status = StatusNotificacao.PENDENTE;

    private LocalDateTime criadaEm = LocalDateTime.now();
    private LocalDateTime encaminhadaEm;

    @Column(nullable = false)
    private int anoReferencia;

    @Column
    private int mesReferencia;
}
